tutorial/source/bayesian_regression.ipynb

```diff
@@ -92,333 +109,367 @@
-      "text/plain": [
-       "<Figure size 864x432 with 2 Axes>"
-      ]
-     },
-     "metadata": {
-      "needs_background": "light"
-     },
      "output_type": "display_data"
     }
    ],
    "source": [
     "fig, ax = plt.subplots(nrows=1, ncols=2, figsize=(12, 6), sharey=True)\n",
-    "african_nations = data[data[\"cont_africa\"] == 1]\n",
-    "non_african_nations = data[data[\"cont_africa\"] == 0]\n",
+    "african_nations = df[df[\"cont_africa\"] == 1]\n",
+    "non_african_nations = df[df[\"cont_africa\"] == 0]\n",
     "sns.scatterplot(non_african_nations[\"rugged\"], \n",
-    "            np.log(non_african_nations[\"rgdppc_2000\"]), \n",
+    "            non_african_nations[\"rgdppc_2000\"], \n",
     "            ax=ax[0])\n",
     "ax[0].set(xlabel=\"Terrain Ruggedness Index\",\n",
     "          ylabel=\"log GDP (2000)\",\n",
     "          title=\"Non African Nations\")\n",
     "sns.scatterplot(african_nations[\"rugged\"], \n",
-    "            np.log(african_nations[\"rgdppc_2000\"]), \n",
-    "            ax=ax[1])\n",
+    "                african_nations[\"rgdppc_2000\"], \n",
+    "                ax=ax[1])\n",
     "ax[1].set(xlabel=\"Terrain Ruggedness Index\",\n",
     "          ylabel=\"log GDP (2000)\",\n",
-    "          title=\"Afri
```

Comment:
not important: I think you can also define `sigma` as PyroSample.

Reply:
This is for the definition of `BayesianRegression` class.

Reply:
That's right, I didn't do that because we don't need to set it as an attribute, and I just wanted to show that we can call regular pyro primitives in the code too, but I should call this out probably.

Reply:
Actually `PyroSample` won't work here.

Reply:
I meant to define `self.sigma = PyroSample(dist.Uniform(0., 10.))` at the module construction. I think that it is a part of the Bayesian NN, so considering it an attribute of BayesianRegression module makes sense to me. Anyway, this is just a minor point to me. :)

Reply:
That seems reasonable to me, will change.

Reply:
Actually, one issue that I see with this pattern is that it is easy to forget that `self.sigma` is a special attribute and we need to access it in the correct context. The current model is:

```python
    def forward(self, x, y=None):
        sigma = pyro.sample("sigma", dist.Uniform(0, 10))
        mean = self.linear(x).squeeze(-1)
        with pyro.plate("data", x.shape[0]):
            obs = pyro.sample("obs", dist.Normal(mean, sigma), obs=y)
        return mean
```

if we change it to have `self.sigma = PyroSample(dist.Uniform(1, 10))` in `__init__` and change the forward method to the following, 

```python
    def forward(self, x, y=None):
        mean = self.linear(x).squeeze(-1)
        with pyro.plate("data", x.shape[0]):
            obs = pyro.sample("obs", dist.Normal(mean, self.sigma), obs=y)
        return mean
```
we'll silently do the unintended thing of actually sampling `sigma` that has a batch size equal to the data, since it is easy to forget that `self.sigma` needs to be accessed in the correct context, i.e. we'll need to add `sigma = self.sigma` outside the plate so as to trigger `getattr` outside the plate context. I'm happy to do this latter thing, but that will need a little bit more explaining in the notebook as to why we did that. I'll make the change based on feedback.

Reply:
Nice observation! This is easy to make mistakes. Thanks for pointing it out!